engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessConcurrencyTest.java

```diff
@@ -220,4 +222,65 @@ public void shouldTerminateXorWithIncident() {
             tuple(BpmnElementType.EVENT_SUB_PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED),
             tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
   }
+
+  @Test
+  // https://github.com/camunda-cloud/zeebe/issues/6587
+  public void shouldInterruptBoundaryEvent() {
+    // given
+    final ProcessBuilder process = Bpmn.createExecutableProcess(PROCESS_ID);
+
+    process
+        .eventSubProcess("event_sub_proc")
+        .startEvent("event_sub_start")
+        .interrupting(true)
+        .message(b -> b.name(MSG_NAME).zeebeCorrelationKeyExpression("key"))
+        .endEvent("event_sub_end");
+
+    final BpmnModelInstance model =
+        process
+            .startEvent("start_proc")
+            .sequenceFlowId("toParallel")
+            .serviceTask("serviceTask", s -> s.zeebeJobType("task"))
+            .boundaryEvent("boundary")
+            .cancelActivity(true)
+            .message(m -> m.name("msg").zeebeCorrelationKeyExpression("key"))
+            .endEvent("end_proc")
+            .done();
+
+    engineRule.deployment().withXmlResource(model).deploy();
+
+    final long processInstanceKey =
+        engineRule
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariables(Map.of("key", 123, "key1", 123))
```

Comment:
❓ The instance is created with two vars: `key` and `key1`, but `key1` appears unused. Did you perhaps make a change to the correlation key expressions? I imagine that both were needed to differentiate the correlations or so.

Reply:
I don't know, I just copied the test back in I didn't write it 😄

Reply:
Might still be a good opportunity to improve this. Perhaps just delete the second var.